src/generate-watermark-info.ts

```diff
@@ -1,44 +1,55 @@
 import { getAll, loadLocal } from '@d2api/manifest/node';
-import { writeFile } from './helpers';
-
+import allWatermarks from '../data/seasons/all-watermarks.json';
+import seasons from '../data/seasons/seasons_unfiltered.json';
+import watermarkToSeason from '../data/seasons/watermark-to-season.json';
+import { diffArrays, writeFile } from './helpers';
 loadLocal();
 
-const debug = false;
+const inventoryItems = getAll('DestinyInventoryItemDefinition');
 
-if (debug) {
-  const watermarks = [
-    ...new Set(
-      getAll('DestinyInventoryItemDefinition')
-        .map(
-          (o) =>
-            o.quality?.displayVersionWatermarkIcons || o.iconWatermark || o.iconWatermarkShelved
-        )
-        .flat()
-        .filter(Boolean)
-    ),
-  ];
+const watermarks = [
+  ...new Set(
+    getAll('DestinyInventoryItemDefinition')
+      .map((o) => o.quality?.displayVersionWatermarkIcons || o.iconWatermark)
+      .flat()
+      .filter(Boolean)
+  ),
+];
 
-  console.log(watermarks, watermarks.length);
-}
+const watermarksShelved = [
+  ...new Set(
+    getAll('DestinyInventoryItemDefinition')
+      .map((o) => o.quality?.displayVersionWatermarkIcons || o.iconWatermarkShelved)
+      .flat()
+      .filter(Boolean)
+  ),
+];
 
-const watermarkHashesSeasons: Record<number, number> = {
-  423789: 1, // Mythos Hack (non-sunset Season 1)
-  4425887: 1, // The Time-Worn Spire (Sunset Season 1)
-  87646118: 2, // Endless Glory
-  235397500: 3, // The Mad Monk
-  89332041: 4, // Cayde's Duds
-  231533811: 5, // Iron Strength
-  915277992: 6, // Warbrick
-  1363705632: 7, // First Light
-  1359616732: 8, // Gambit Emerald (Undying)
-  2056256564: 8, // Lunar Halcyon Gilt (Shadowkeep)
-  980059630: 9, // Vitrified Chronology
-  737170669: 10, // Valkyrie Zero
-  51755992: 11, // Throne of Soot
-  1371145728: 12, // Amethyst Bloom (Hunt)
-  1948818058: 12, // Gilded Smoke (Beyond Light)
-  117960780: 13, // Peat Bog Boogie (Chosen)
-};
+const watermarksAndShelved = watermarks.concat(watermarksShelved);
```

Comment:
would love a better way to do this automatically, but was missing some shelved icons when doing
```
o.quality?.displayVersionWatermarkIcons || o.iconWatermark || o.iconWatermarkShelved
```